### PR TITLE
Improve readme to include experimental features

### PR DESCRIPTION
**Pull request by jabraham17**
Improve readme to include experimental features

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # chapel-vscode extension
 
-## Features
+## Core Features
 
 - Intellisense
-  - GotoDefinition: jump to a symbols definition
+  - GotoDefinition: jump to a symbol's definition
```
Comment (ShreyasKhandekar) [inline README.md:6]:
```suggestion
  - Go to Definition: jump to or peek a symbol's definition
```

File: README.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
   - Hover: hover over a symbol to view its signature and documentation
   - Code Completion: suggest completion for symbols in the current file
   - Symbol List: see all symbols for a file
```
Comment (ShreyasKhandekar) [inline README.md:9]:
```suggestion
  - Symbol List: see all symbols in a file
```
Not sure if "for a file" means something different than what I understood, but "in" seems to be the better wording here.

Comment (ShreyasKhandekar) [inline README.md:9]: We also have the ability to jump to or peek all references for a certain symbol right? I think Daniel had demoed that at some point. We should add that in here too.

File: README.md [hunk 3]
```diff
@@ -13,9 +13,16 @@
 - Snippets
 - Syntax Highlighting
 
+### Experimental Features
```
Comment (ShreyasKhandekar) [inline README.md:16]: Should we tell users how to enable these experimental features?

**Review by ShreyasKhandekar — APPROVED** [review]
Thanks for Updating this Jade!
In terms of experimental features I had a small comment about adding a part to tell users how to opt in. I think for the short term this is okay as is.

In the longer term I believe we should not have experimental features published to users for the stable release, rather we should use the VS Code Marketplace's ability to have a Beta/pre-release version of the extension to allow users who are interested to opt in to these features, till they are stable enough for the stable release.

File: README.md (hunk 3 above)
Comment (jabraham17) [reply README.md:16]: Yes, I'll add a note about enabling the resolver (with the caveat that it only works with limited Chapel code)

**Comment by jabraham17** [comment]
> In the longer term I believe we should not have experimental features published to users for the stable release, rather we should use the VS Code Marketplace's ability to have a Beta/pre-release version of the extension to allow users who are interested to opt in to these features, till they are stable enough for the stable release.

While to some extent I agree with that, its not within the control of the extension for when these features are available. The "experimental features" are fully baked language server features, its just that the dyno resolver needs to be more feature complete before these are on by default

File: README.md (hunk 2 above)
Comment (jabraham17) [reply README.md:9]: Done!

**Comment by ShreyasKhandekar** [comment]
Okay I see, in that case I was misunderstanding the experimental nature of the features. I think we can keep them this way in that case.

**Pull request merged**